tests/common/glcubetexturelayerrenderer.cpp

```diff
@@ -0,0 +1,622 @@
+/*
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "glcubetexturelayerrenderer.h"
+#include <unistd.h>
+#include <sys/mman.h>
+#include <cassert>
+#include <cstring>
+#include <iostream>
+
+struct Dimension {
+    EGLint width = 0;
+    EGLint height = 0;
+    EGLint stride = 0;
+};
+
+class StreamTextureImpl
+{
+public:
+   ~StreamTextureImpl ()
+  {
+    glDeleteTextures (1, &gl_tex_);
+    gl_->eglDestroyImageKHR (gl_->display, image_);
+    close (fd_);
+    gbm_bo_destroy (bo_);
+  }
+
+  void *Map ()
+  {
+    assert (addr_ == nullptr);
+    size_t size = dimension_.stride * dimension_.height;
+    addr_ =
+      mmap (nullptr, size, (PROT_READ | PROT_WRITE), MAP_SHARED, fd_, 0);
```

Comment:
Another good example would be software decoded video. In that example, we wouldnt involve any 3D side at all here, can u extend this to include it ?